tests/utils/testutils.py

```diff
@@ -287,16 +308,18 @@ def _run_test(self, test, *args, **kwargs):
                 test(self, *args, **kwargs)
             except Exception as e:
                 test_exception = e
-
-            try:
-                self.host_stdout.seek(last_pos)
-                self.host_out = self.host_stdout.read()
-                self.host_stdout_logger.error(
-                    'Captured WebHost stdout from %s :\n%s',
-                    self.host_stdout.name, self.host_out)
             finally:
-                if test_exception is not None:
-                    raise test_exception
+                try:
+                    self.host_stdout.seek(last_pos)
+                    self.host_out = self.host_stdout.read()
+                    if self.host_out is not None and len(self.host_out) > 0:
+                        self.host_stdout_logger.error(
```

Comment:
Would this also help that the webhost logs file to look at the is also listed?

Reply:
probably not, archiving webhost logs happens in teardown of every test capturing complete webhost log from tmpfile. here is more for not archiving case where we want to log webhost log segment happening for failed tests.